Button Override and Cleaning Subsystem

Robin27182 opened the pull request:
Added feature to override the min and max of the lift positions. Changed LiftPidControl to account for this change. Please check RobotContainer that I am calling zl and zr correctly to have them update.

Drastically cleaned up Lift subsystem, (it was messy and it helps me navigate.) Will be able to alter code from 3 - 5 tomorrow or right after the meeting

Robin27182 committed:
Added feature to override the min and max of the lift positions. Changed LiftPidControl to account for this change. Please check RobotContainer that I am calling zl and zr correctly to have them update.

Drastically cleaned up lift subsystem. (It helps me navigate, and I didn't think it would be bad) I will be able to make changes tomorrow directly after the meeting or 3 - 5 tomorrow.

clockworkthought commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -48,7 +48,8 @@ public RobotContainer() {
         lift = new LiftSubsystem(driveSubsystem.chassis::SetAccelerationLimit, coralSubsystem, algaeSubsystem);
-        lift.setDefaultCommand(new LiftPidControl(lift, () -> payloadController.getLeftY()));
+        lift.setDefaultCommand(new LiftPidControl(lift, () -> payloadController.getLeftY(),
+                payloadController.zl().getAsBoolean() && payloadController.zr().getAsBoolean()));
```
Because it's a boolean being evaluated here, rather than a boolean supplier, the LifdPidControl command will run ````payloadController.zl().getAsBoolean() && payloadController.zr().getAsBoolean())````, store it as its value at the time RobotContainer is constructed (which will almost always be ````false```), then create the LiftPidControl command and pass the ````false```` or ```true```` in, and then never update it later.

clockworkthought commented on src/main/java/frc/robot/subsystems/LiftSubsystem.java:
```diff
@@ -21,9 +16,15 @@ public class LiftSubsystem extends SubsystemBase {
+
+
+    final double CHANGE_MULTIPLIER = 2.2;
+    final double MAX_SPEED_PRECENT = .6;
```
Java doesn't care, but "PERCENT", rather than "PRECENT".

Robin27182 committed:
Added feature to override the min and max of the lift positions. Changed LiftPidControl to account for this change. Please check RobotContainer that I am calling zl and zr correctly to have them update.

Drastically cleaned up lift subsystem. (It helps me navigate, and I didn't think it would be bad) I will be able to make changes tomorrow directly after the meeting or 3 - 5 tomorrow.

Robin27182 commented:
I think I fixed it. I passed in the payloadController.zl() and the other for zr, and I called the .getAsBoolean in the PidControl for the lift.

clockworkthought committed:
Further bug fixes

clockworkthought committed:
Cleanup commit,  additional AprilAlignment work.

clockworkthought committed:
Cleanup commit,  additional AprilAlignment work.

clockworkthought committed:
Fixes to turn_magnitude for directly forward or backward.

clockworkthought committed:
Fixes to turn_magnitude for directly forward or backward.
